src/workerd/api/eventsource.c++

```diff
@@ -307,7 +307,7 @@ void EventSource::notifyMessages(jsg::Lock& js, kj::Array<PendingMessage> messag
     for (auto& message: messages) {
       auto data = kj::str(kj::delimited(kj::mv(message.data), "\n"_kjc));
       if (data.size() == 0) continue;
-      kj::String type = kj::mv(message.event).orDefault(kj::str("message"));
+      kj::String type = kj::mv(message.event).orDefault([]() { return kj::str("message"); });
```

Comment:
I believe the `MessageEvent` constructor can take a `kj::StringPtr` for the second argument. If that's the case, it might be worthwhile to avoid the allocation here entirely with something like...

```cpp
kj::StringPtr type = "message"_kjc;
KJ_IF_SOME(event, message.event) {
  type = event.asPtr();
}
```